server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -384,12 +386,16 @@ public Query nullValueQuery() {
 
     public abstract Query existsQuery(QueryShardContext context);
 
-    public Query phraseQuery(String field, TokenStream stream, int slop, boolean enablePositionIncrements) throws IOException {
-        throw new IllegalArgumentException("Can only use phrase queries on text fields - not on [" + name + "] which is of type [" + typeName() + "]");
+    public Query phraseQuery(String field, TokenStream stream, int slop, boolean enablePositionIncrements,
+                             DeprecationLogger deprecationLogger) throws IOException {
+        deprecationLogger.deprecated("Attempted to build a phrase query with multiple terms against non-text field [{}]", name);
+        return new MatchNoDocsQuery();
     }
 
-    public Query multiPhraseQuery(String field, TokenStream stream, int slop, boolean enablePositionIncrements) throws IOException {
-        throw new IllegalArgumentException("Can only use phrase queries on text fields - not on [" + name + "] which is of type [" + typeName() + "]");
+    public Query multiPhraseQuery(String field, TokenStream stream, int slop, boolean enablePositionIncrements,
```

Comment:
we could use a static deprecation logger instead of requiring callers to pass one?